core/include/detray/materials/material_slab.hpp

```diff
@@ -51,6 +55,21 @@ struct material_slab {
     /// Return the nuclear interaction length fraction.
     DETRAY_HOST_DEVICE
     constexpr scalar_type thickness_in_L0() const { return m_thickness_in_L0; }
+    /// Return the interaction length
+    DETRAY_HOST_DEVICE
+    scalar_type interaction_length(const line_plane_intersection& is) const {
+        return m_thickness / is.cos_incidence_angle;
```

Comment:
But does this work for a cylinder?

Reply:
I think you are concerning the curvature of cylinder surface, but I guess that's not a very critical factor.

Reply:
> E.g. for the line, you look at the cord right now, but what if the track is almost parallel to the line in z direction?
You are right. I fixed the code to consider the incidence angle for line intersection.

> Still, we might be better off using a material rod for a cylinder? It is very similar to the circular line, I guess
It will be better. But let's just use the slab for the moment as ACTS does. It's not that trivial to implement it with rod.